core/xml.js

```diff
@@ -86,6 +86,30 @@ Blockly.Xml.blockToDomWithXY = function(block, opt_noId) {
   return element;
 };
 
+Blockly.Xml.fieldToDomVariable_ = function(field, workspace) {
+  var id = field.getValue();
+  var variable = workspace.getVariableById(id);
+  if (!variable) {
+    if (workspace.isFlyout && workspace.targetWorkspace) {
```

Comment:
do you need to check for workspace.targetWorkspace here? You don't use it in the conditional.

Reply:
Whoops, I missed that one when I hit merge.  I'll do this in a followup.